components/centraldashboard/app/api.ts

```diff
@@ -3,72 +3,100 @@ import express from 'express';
 import {KubernetesService, PlatformInfo} from './k8s_service';
 import {Interval, MetricsService} from './metrics_service';
 
-interface UserInfo {
-  email: string;
-  image: string;
-}
-
 interface EnvironmentInfo {
-  user: UserInfo;
+  namespaces: string[];
   platform: PlatformInfo;
+  user: string;
 }
 
-/** Function that provides user information back */
-function getUserInfo() {
-  return {
-    email: 'user@kubeflow.org',
-  };
-}
+const IAP_HEADER = 'X-Goog-Authenticated-User-Email';
+const IAP_PREFIX = 'accounts.google.com:';
+
+export class Api {
+  private platformInfo: PlatformInfo;
 
-/** API routes exposing information from Kubernetes */
-export function api(
-    k8sService: KubernetesService,
-    metricsService?: MetricsService): express.Router {
-  return express.Router()
-      .get(
-          '/env-info',
-          async (_: express.Request, res: express.Response) => {
-            const [platform, user] = await Promise.all([
-              k8sService.getPlatformInfo(),
-              getUserInfo(),
-            ]);
-            const returnPayoad = {platform, user} as EnvironmentInfo;
-            res.json(returnPayoad);
-          })
-      .get(
-          '/metrics/:type((node|podcpu|podmem))',
-          async (req: express.Request, res: express.Response) => {
-            let interval = Interval.Last15m;
-            if (Interval[req.query.interval] !== undefined) {
-              interval = Number(Interval[req.query.interval]);
-            }
-            if (metricsService) {
-              switch (req.params.type) {
-                case 'node':
-                  res.json(
-                      await metricsService.getNodeCpuUtilization(interval));
-                  break;
-                case 'podcpu':
-                  res.json(await metricsService.getPodCpuUtilization(interval));
-                  break;
-                case 'podmem':
-                  res.json(await metricsService.getPodMemoryUsage(interval));
-                  break;
-                default:
+  constructor(
+      private k8sService: KubernetesService,
+      private metricsService?: MetricsService) {}
+
+  /** Retrieves and memoizes the PlatformInfo. */
+  private async getPlatformInfo(): Promise<PlatformInfo> {
+    if (!this.platformInfo) {
+      this.platformInfo = await this.k8sService.getPlatformInfo();
+    }
+    return this.platformInfo;
+  }
+
+  /**
+   * Retrieves user information from headers.
+   * Supports:
+   *  GCP IAP (https://cloud.google.com/iap/docs/identity-howto)
+   */
+  private getUser(req: express.Request): string {
+    let email = 'anonymous@kubeflow.org';
```

Comment:
_**nit**_ Code depth
```suggestion
    const iap = req.header(IAP_HEADER)
    return iap? iap.slice(IAP_PREFIX.length) : 'anonymous@kubeflow.org';
```

Reply:
I was expecting that eventually there might be additional headers that would be interrogated to support other authentication providers besides IAP.

Reply:
I see, sure, we can leave it as it is.